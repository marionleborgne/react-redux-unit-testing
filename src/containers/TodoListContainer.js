import React from 'react';
import { connect } from 'react-redux';
import TodoListComponent from '../components/TodoListComponent';
import { addTodo, toggleTodo } from '../actions';

class TodoList extends React.Component {

  _onItemClick = item => this.props.toggleTodo(item);

  _createItem = text => this.props.addTodo(text);

  render() {
    return <TodoListComponent
      onItemClick={this._onItemClick}
      createItem={this._createItem}
      items={this.props.items}
    />
  }
}

export const mapStateToProps = state => ({
  items: state.todos
});

const mapDispatchToProps = dispatch => ({
  addTodo: text => dispatch(addTodo(text)),
  toggleTodo: item => dispatch(toggleTodo(item.id))
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);

