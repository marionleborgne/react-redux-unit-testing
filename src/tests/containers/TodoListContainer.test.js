import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import TodoListContainer, { mapStateToProps } from '../../containers/TodoListContainer';

const initialState = {
  todos: [
    {
      text: 'Test',
      id: 'mock',
      done: true
    }
  ]
};
const initialProps = mapStateToProps(initialState);
const mockStore = configureStore();

describe('TodoListContainer', () => {

  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(
      <TodoListContainer.WrappedComponent {...initialProps} />,
      { context: { store } }
    );
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('renders a shallow copy with expected initial state', () => {
    const props = wrapper.props();
    const actual = props.items;
    const expected = initialState.todos;
    expect(actual).toEqual(expected);
  });

  it('renders with expected initial state', () => {
    const wrapper = mount(<Provider store={store}><TodoListContainer /></Provider>);
    const container = wrapper.find(TodoListContainer.WrappedComponent)
    const actual = container.props().items;
    const expected = initialState.todos;
    expect(actual).toEqual(expected);
  });
});
