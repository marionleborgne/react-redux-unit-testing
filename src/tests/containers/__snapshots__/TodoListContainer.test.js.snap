// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListContainer renders 1`] = `
<_default
  createItem={[Function]}
  items={
    Array [
      Object {
        "done": true,
        "id": "mock",
        "text": "Test",
      },
    ]
  }
  onItemClick={[Function]}
/>
`;
